internal/engine/wazevo/ssa/pass.go

```diff
@@ -348,6 +339,88 @@ func passNopInstElimination(b *builder) {
 						b.alias(cur.Return(), x)
 					}
 				}
+			// Z := Const 0
+			// (Iadd X, Z) => X
+			// (Iadd Z, Y) => Y
+			case OpcodeIadd:
+				x, y := cur.Arg2()
+				definingInst := b.valueIDToInstruction[y.ID()]
+				if definingInst == nil {
+					if definingInst = b.valueIDToInstruction[x.ID()]; definingInst == nil {
+						continue
+					} else {
+						x = y
+					}
+				}
+				if definingInst.Constant() && definingInst.ConstantVal() == 0 {
+					b.alias(cur.Return(), x)
+				}
+			}
+		}
+	}
+}
+
+func ensureValueIdToInstructionInit(b *builder) {
+	b.valueIDToInstruction = b.valueIDToInstruction[:0]
```

Comment:
as I said this initializes and iterates over instructions per invocation of this function.

Reply:
should be fixed now